codex-rs/core/src/shell.rs

```diff
@@ -116,13 +116,45 @@ fn strip_bash_lc(command: &Vec<String>) -> Option<String> {
     }
 }
 
+fn get_zsh_shell() -> Option<ZshShell> {
+    use libc::getpwuid;
+    use libc::getuid;
+    use std::ffi::CStr;
+
+    unsafe {
+        let uid = getuid();
+        let pw = getpwuid(uid);
+
+        if !pw.is_null() {
+            let shell_path = CStr::from_ptr((*pw).pw_shell)
+                .to_string_lossy()
+                .into_owned();
+
+            if shell_path.contains("zsh") {
```

Comment:
Shouldn't we also check for `Bash` here (assuming we add `Bash` as a variant to `Shell`) and then this function should return `Shell`? Instead of `None`, it can return `Shell::Unknown`?